plugins/catalog-backend/src/ingestion/processors/ldap/config.ts

```diff
@@ -0,0 +1,264 @@
+import { RecursivePartial } from './util';
+
+/**
+ * The configuration parameters for a single LDAP provider.
+ */
+export type LdapProviderConfig = {
+  // The prefix of the target that this matches on, e.g.
+  // "ldaps://ds.example.net", with no trailing slash.
+  target: string;
+  // The settings to use for the bind command. If none are specified, the bind
+  // command is not issued.
+  bind?: BindConfig;
+  // The settings that govern the reading and interpretation of users
+  users: UserConfig;
+  // The settings that govern the reading and interpretation of groups
+  groups: GroupConfig;
+};
+
+/**
+ * The settings to use for the a command.
+ */
+export type BindConfig = {
+  // The DN of the user to auth as, e.g.
+  // uid=ldap-robot,ou=robots,ou=example,dc=example,dc=net
+  dn: string;
+  // The secret of the user to auth as (its password)
+  secret: string;
+};
+
+/**
+ * The settings that govern the reading and interpretation of users.
+ */
+export type UserConfig = {
+  // The DN under which users are stored.
+  dn: string;
+  // The search options to use.
+  // Only the scope, filter, attributes, and paged fields are supported. The
+  // default is scope "one" and attributes "*" and "+".
+  options: SearchOptions;
+  // JSON paths (on a.b.c form) and hard coded values to set on those paths
+  set?: { path: string; value: JsonValue }[];
+  // Mappings from well known entity fields, to LDAP attribute names
+  map: {
+    // The name of the attribute that holds the relative distinguished name of
+    // each entry. Defaults to "uid".
+    rdn: string;
+    // The name of the attribute that shall be used for the value of the
+    // metadata.name field of the entity. Defaults to "uid".
+    name: string;
+    // The name of the attribute that shall be used for the value of the
+    // metadata.description field of the entity.
+    description?: string;
+    // The name of the attribute that shall be used for the value of the
+    // spec.profile.displayName field of the entity. Defaults to "cn".
+    displayName: string;
+    // The name of the attribute that shall be used for the value of the
+    // spec.profile.email field of the entity. Defaults to "mail".
+    email: string;
+    // The name of the attribute that shall be used for the value of the
+    // spec.profile.picture field of the entity.
+    picture?: string;
+    // The name of the attribute that shall be used for the values of the
+    // spec.memberOf field of the entity. Defaults to "memberOf".
+    memberOf: string;
+  };
+};
+
+/**
+ * The settings that govern the reading and interpretation of groups.
+ */
+export type GroupConfig = {
+  // The DN under which groups are stored.
+  dn: string;
+  // The search options to use.
+  // Only the scope, filter, attributes, and paged fields are supported.
+  options: SearchOptions;
+  // JSON paths (on a.b.c form) and hard coded values to set on those paths
+  set?: { path: string; value: JsonValue }[];
```

Comment:
Is this our own invention, what's the idea in that case?

Reply:
Yeah. So LDAP is utterly free-form, and our schema has required fields. There will be cases where the input data cannot possibly fulfil our output schema. Main thing is for example filling in groups' `spec.type` with a hard coded value such as `"team"`.